### PR TITLE
Generate all n balanced parenthesis combinations

### PR DESCRIPTION
**Pull request by gayatri517**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

Added generate valid parenthesis file in cp(competitive programming) folder.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#4023

### CODE REVIEW

**Comment by gayatri517** [comment]
@HarshCasper kindly review it.

File: C-Plus-Plus/cp/generate_valid_parenthesis.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*Problem Statement: 
+You are given a number n, your job is to find al the n balanced parenthesis
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/generate_valid_parenthesis.cpp]: It should be `all` not `al`

File: C-Plus-Plus/cp/generate_valid_parenthesis.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+combinations possible */
+
+#include <bits/stdc++.h>
+using namespace std;
+
+void FindValidParenthesis(char *out, int n, int index, int open, int close)
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/generate_valid_parenthesis.cpp:19]: Add a function description

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by gayatri517** [comment]
@HarshCasper I have made the changes. Kindly review.
Thank you.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**